CONTRIBUTING.md

```diff
@@ -0,0 +1,246 @@
+  or feature implementation. **Do not** refactor or reformat code that is
+  unrelated to your change. It is better to **submit many small pull requests**
+  rather than a single large one. Enormous pull requests will take enormous
+  amounts of time to review, or may be rejected altogether.
+
+- **Coordinate bigger changes.** For large and non-trivial changes, open an
+  issue to discuss a strategy with the maintainers. Otherwise, you risk doing a
+  lot of work for nothing!
+
+- **Prioritize understanding over cleverness.** Write code clearly and
+  concisely. Remember that source code usually gets written once and read
+  often. Ensure the code is clear to the reader. The purpose and logic should
+  be obvious to a reasonably skilled developer, otherwise you should add a
+  comment that explains it.
+
+- **Follow existing coding style and conventions.** Keep your code consistent
+  with the style, formatting, and conventions in the rest of the code base.
+  When possible, these will be enforced with a linter. Consistency makes it
+  easier to review and modify in the future.
+
+- **Include test coverage.** Add unit tests or UI tests when possible. Follow
+  existing patterns for implementing tests.
+
+- **Update the example project** if one exists to exercise any new
+  functionality you have added.
+
+- **Add documentation.** Document your changes with code doc comments or in
+  existing guides.
+
+- **Use the repository default branch.** Branch from and [submit your pull
+  request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)
+  to the repository default branch. Usually this is `main`, but it could be
+  `dev`, `develop`, or `master`.
+
+- **[Resolve any merge
+  conflicts](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-on-github)**
+  that occur.
+
+- **Promptly address any CI failures**. If your pull request fails to build or
+  pass tests, please push another commit to fix it.
+
+- **Use [`pre-commit`][pre-commit] whenever possible**. We want to provide high
+  quality code. For this reason some of our repositories are using several
+  "pre-commit hooks". Please check your code before you will create
+  pull-requests. See [pre-commit documentation][pre-commit] for further
+  details.
+
+- When writing comments, use properly constructed sentences, including
+  punctuation.
+
+## :memo: Writing Commit Messages
+
+Depending on the project we tend to use various approaches for writing good
+commit messages. See [this](https://chris.beams.io/posts/git-commit/) as an
+example of a "good" commits message. We also tend to use [Conventional
+Commits][conventional], with a scope detailing the area of the change being
+made.
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
+3. Capitalize the subject line
```

Comment:
Why this is included?

> 3. Capitalize the subject line

e.g. I prefer this: `docs: fix typo` over this `docs: Fix typo`

Reply:
Same here, I stick to the former one. However, I see where the latter comes from: mixing [Chris Beam's suggestion](https://cbea.ms/git-commit/#capitalize) with conventional commits.

I think both @kamilzyla and @jakubnowicki use the latter version.

Reply:
@DamianBudelewski @tmakowski, I've rephrased this section a bit and made capitalization point optional. I think we should not enforce specific format (either the one suggested by Chris Beams or Conventional Commit or any other) as long as the repository consistency is preserved and commit messages meet basic requirements, suit their original purpose and are human readable.